hazelcast/src/main/java/com/hazelcast/query/impl/QueryableEntry.java

```diff
@@ -171,9 +186,6 @@ private static AttributeType extractAttributeTypeFromSingleResult(Object extract
         if (extractedSingleResult == null) {
             return null;
         }
-        if (extractedSingleResult instanceof HazelcastJsonValue) {
```

Comment:
I'm just a bit lost with the extractors here and different types going around (`HazelcastJsonValue` vs `JsonValue`). Also, `QueryableEntry#extractAttributeTypeFromJsonValue` is now unused. Can you tell me why we don't need this anymore and what is the attribute value type and attribute type in the case we are handling JSON?

Reply:
To be honest, type conversions part is quite complicated. I still dont understand it wholly either.
The querying system only returns a HazelcastJsonValue when it handles a `this` query. For other queries, AbstractJsonGetter already returns a primitive. We convert HazelcastJsonValue to JsonValue inside `#getAttributeValue` now. Then, this JsonValue is converted to whatever `Comparable` type it is compared against inside `AbstractPredicate#convertAndApplyForSingleAttributeValue`. Therefore, no JsonValue or HazelcastJsonValue reaches this part. So I removed this part.

Reply:
Ok, thanks!